In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

## Reading data from csv files 
df= pd.read_csv('noshowappointments-kagglev2-may-2016.csv')

## Showing data in the file  
df.head()


PatientId  AppointmentID Gender          ScheduledDay  \
0  2.987250e+13        5642903      F  2016-04-29T18:38:08Z   
1  5.589978e+14        5642503      M  2016-04-29T16:08:27Z   
2  4.262962e+12        5642549      F  2016-04-29T16:19:04Z   
3  8.679512e+11        5642828      F  2016-04-29T17:29:31Z   
4  8.841186e+12        5642494      F  2016-04-29T16:07:23Z   

         AppointmentDay  Age      Neighbourhood  Scholarship  Hipertension  \
0  2016-04-29T00:00:00Z   62    JARDIM DA PENHA            0             1   
1  2016-04-29T00:00:00Z   56    JARDIM DA PENHA            0             0   
2  2016-04-29T00:00:00Z   62      MATA DA PRAIA            0             0   
3  2016-04-29T00:00:00Z    8  PONTAL DE CAMBURI            0             0   
4  2016-04-29T00:00:00Z   56    JARDIM DA PENHA            0             1   

   Diabetes  Alcoholism  Handcap  SMS_received No-show  
0         0           0        0             0      No  
1         0           0        0             0      No  
2         0           0        0             0      No  
3         0           0        0             0      No  
4         1           0        0             0      No

In [3]:
## Viewing data formats for rows and misisng data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110527 entries, 0 to 110526
Data columns (total 14 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   PatientId       110527 non-null  float64
 1   AppointmentID   110527 non-null  int64  
 2   Gender          110527 non-null  object 
 3   ScheduledDay    110527 non-null  object 
 4   AppointmentDay  110527 non-null  object 
 5   Age             110527 non-null  int64  
 6   Neighbourhood   110527 non-null  object 
 7   Scholarship     110527 non-null  int64  
 8   Hipertension    110527 non-null  int64  
 9   Diabetes        110527 non-null  int64  
 10  Alcoholism      110527 non-null  int64  
 11  Handcap         110527 non-null  int64  
 12  SMS_received    110527 non-null  int64  
 13  No-show         110527 non-null  object 
dtypes: float64(1), int64(8), object(5)
memory usage: 11.8+ MB


From information about Nowhowappointements-kagglev2-may-2016 there are no missing dat for the analyis

In [7]:
## Getting the statistcal summaries using the describe function. 
df.describe()

PatientId  AppointmentID            Age    Scholarship  \
count  1.105270e+05   1.105270e+05  110527.000000  110527.000000   
mean   1.474963e+14   5.675305e+06      37.088874       0.098266   
std    2.560949e+14   7.129575e+04      23.110205       0.297675   
min    3.921784e+04   5.030230e+06      -1.000000       0.000000   
25%    4.172614e+12   5.640286e+06      18.000000       0.000000   
50%    3.173184e+13   5.680573e+06      37.000000       0.000000   
75%    9.439172e+13   5.725524e+06      55.000000       0.000000   
max    9.999816e+14   5.790484e+06     115.000000       1.000000   

        Hipertension       Diabetes     Alcoholism        Handcap  \
count  110527.000000  110527.000000  110527.000000  110527.000000   
mean        0.197246       0.071865       0.030400       0.022248   
std         0.397921       0.258265       0.171686       0.161543   
min         0.000000       0.000000       0.000000       0.000000   
25%         0.000000       0.000000       0.000000       0.000000   
50%         0.000000       0.000000       0.000000       0.000000   
75%         0.000000       0.000000       0.000000       0.000000   
max         1.000000       1.000000       1.000000       4.000000   

        SMS_received  
count  110527.000000  
mean        0.321026  
std         0.466873  
min         0.000000  
25%         0.000000  
50%         0.000000  
75%         1.000000  
max         1.000000

In [8]:
## How many patients showed up after booking appointment ? HOw many patients didn't show up after booking appointment?
## Getting the total number of those who showd up for the appointment and those who didn't

df['No-show'].value_counts()

No     88208
Yes    22319
Name: No-show, dtype: int64

A total number of 88,208 of the patient who booked for appointment showed up while 22,319 patinets didnt show up after booking appointment 

In [39]:
## Which age group booked the highest number of appointiments ? 
df_age= df.groupby(df['Age']).count()
df_age

PatientId  AppointmentID  Gender  ScheduledDay  AppointmentDay  \
Age                                                                    
-1            1              1       1             1               1   
 0         3539           3539    3539          3539            3539   
 1         2273           2273    2273          2273            2273   
 2         1618           1618    1618          1618            1618   
 3         1513           1513    1513          1513            1513   
...         ...            ...     ...           ...             ...   
 98           6              6       6             6               6   
 99           1              1       1             1               1   
 100          4              4       4             4               4   
 102          2              2       2             2               2   
 115          5              5       5             5               5   

      Neighbourhood  Scholarship  Hipertension  Diabetes  Alcoholism  Handcap  \
Age                                                                             
-1                1            1             1         1           1        1   
 0             3539         3539          3539      3539        3539     3539   
 1             2273         2273          2273      2273        2273     2273   
 2             1618         1618          1618      1618        1618     1618   
 3             1513         1513          1513      1513        1513     1513   
...             ...          ...           ...       ...         ...      ...   
 98               6            6             6         6           6        6   
 99               1            1             1         1           1        1   
 100              4            4             4         4           4        4   
 102              2            2             2         2           2        2   
 115              5            5             5         5           5        5   

      SMS_received  No-show  
Age                          
-1               1        1  
 0            3539     3539  
 1            2273     2273  
 2            1618     1618  
 3            1513     1513  
...            ...      ...  
 98              6        6  
 99              1        1  
 100             4        4  
 102             2        2  
 115             5        5  

[104 rows x 13 columns]